src/pytorch_lightning/strategies/colossalai.py

```diff
@@ -463,8 +463,17 @@ def broadcast(self, obj: TBroadcast, src: int = 0) -> TBroadcast:
         with _patch_cuda_is_available():
             from colossalai.communication.collective import broadcast
             from colossalai.context import ParallelMode
+            from colossalai.core import global_context as gpc
 
-        return broadcast(obj, src=src, parallel_mode=ParallelMode.GLOBAL)
+        if isinstance(obj, Tensor):
+            return broadcast(obj, src=src, parallel_mode=ParallelMode.GLOBAL)
+        else:
+            obj_list = [obj]
+            local_rank = gpc.get_local_rank(ParallelMode.GLOBAL)
+            if local_rank != src:
+                obj_list = [None]  # type: ignore[list-item]
```

Comment:
Do we need to do this? According to `torch.distributed` [docs](https://pytorch.org/docs/stable/distributed.html#torch.distributed.broadcast_object_list), only data from `src` will be actually broadcast and rest is discarded.

Reply:
ok. I'll check it.